src/Riok.Mapperly/Descriptors/MappingBuilder/NewInstanceMappingBuilderContext.cs

```diff
@@ -17,6 +17,9 @@ public void AddInitPropertyMapping(PropertyAssignmentMapping mapping)
 
     public void AddConstructorParameterMapping(ConstructorParameterMapping mapping)
     {
+        if (PropertyConfigsByRootTargetName.ContainsKey(mapping.Parameter.Name))
```

Comment:
I think the contains key call is not needed at all, since the remove is a no-op if the key isn't found.